tgui/packages/tgui/interfaces/CrewConsole.jsx

```diff
@@ -86,25 +87,119 @@ export const CrewConsole = () => {
 
 const CrewTable = (props) => {
   const { act, data } = useBackend();
-  const sensors = sortBy(data.sensors ?? [], (s) => s.ijob);
+  const sensors = data.sensors ?? [];
+
+  const [shownSensors, setShownSensors] = useState(sensors.slice());
+  const [sortColumn, setSortColumn] = useState('ijob');
+  const [sortAsc, setSortAsc] = useState(true);
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const updateSortMode = (column) => {
+    if (column === sortColumn) {
+      if (!sortAsc) {
+        setSortColumn('ijob');
+        setSortAsc(true);
+      } else {
+        setSortAsc(false);
+      }
+    } else {
+      setSortColumn(column);
+      setSortAsc(true);
+    }
+  };
+
+  const healthSort = (a, b) => {
+    if (a.life_status < b.life_status) return -1;
+    if (a.life_status > b.life_status) return 1;
+    if (a.health > b.health) return -1;
+    if (a.health < b.health) return 1;
+    return 0;
+  };
+
+  useEffect(() => {
+    let unsorted = sensors.slice();
+    if (sortColumn === 'ijob') {
+      let sorted = sortBy(data.sensors ?? [], (s) => s.ijob).filter((crew) =>
+        crew.name.toLowerCase().includes(searchQuery.toLowerCase()),
+      );
```

Comment:
data.sensors will never be null as per earlier,

you are using the `sortBy` fn but standard lib `filter`, this should be made a flow with `sortBy(sensors, ` and `filter(crew, ` 

 also use `createSearch` please

Reply:
> you are using the sortBy fn but standard lib filter, this should be made a flow with sortBy(sensors, and filter(crew, 

Will do, but I gotta ask, why use these custom-made functions instead of the built-in ones? I'm guessing it has to do with browser compatibility?

Reply:
setting my opinion aside: implementations from `collections` are generally faster, they are basically lodash equivalent